doc/classes/ImageTexture.xml

```diff
@@ -4,10 +4,38 @@
 		A [Texture2D] based on an [Image].
 	</brief_description>
 	<description>
-		A [Texture2D] based on an [Image]. Can be created from an [Image] with [method create_from_image].
-		[b]Note:[/b] The maximum image size is 16384×16384 pixels due to graphics hardware limitations. Larger images will fail to import.
+		A [Texture2D] based on an [Image]. For an image to be displayed, an [ImageTexture] has to be created from it using the [method create_from_image] method:
+		[codeblock]
+		var texture = ImageTexture.new()
+		var image = Image.new()
+		image.load("res://icon.png")
+		texture.create_from_image(image)
+		$Sprite2D.texture = texture
+		[/codeblock]
+		This way, textures can be created at run-time by loading images both from within the editor and externally.
+		[b]Warning:[/b] Prefer to load imported textures over loading them from within the filesystem dynamically, as it may not work in exported projects:
+		[codeblock]
+		var texture = load("res://icon.png")
+		$Sprite2D.texture = texture
+		[/codeblock]
+		This is because [method Image.load] relies on OS-specific file access methods that bypass the virtual filesystem implemented in the engine to load imported resources, so [method Image.load] should only be used in the editor or in cases when you need to load external images at run-time, such as images located at the [code]user://[/code] directory.
+		If you'd still like to load an image file just like any other [Resource], import it as an [Image] resource instead, and then load it normally using the [method @GDScript.load] method:
+		[codeblock]
+		var image = load("res://icon.png")
+		assert(image is Image)
+		assert(not (image is ImageTexture))
+		[/codeblock]
+		But do note that the image data can still be retrieved from an imported texture as well using the [method Texture2D.get_data] method, which returns a copy of the data:
+		[codeblock]
+		var texture = load("res://icon.png")
+		var image = texture.get_data()
+		assert(image is Image)
```

Comment:
```suggestion
		var image : Image = texture.get_data()
```
IMO type hints are a better way to convey type information than asserts.

Reply:
I just recall you weren't happy with including type hints in documentation. And I think `assert()` works better for detecting potential bugs if users copy-paste code snippets. This is more like doctest in Python, but that's up for a proposal!

Reply:
Well `Texture2D.get_data()`'s return type is `Image`, so the `assert()` here is not useful, as the type cannot be something else than `Image`. `image.is_empty()` might be closer to the kind of sanity checks that users could do, but that's also outside the scope of this documentation IMO.

Type hints are not used currently in the docs, but if they make sense in specific code blocks, they can be used.

Otherwise another way is:
```
# Texture2D.get_data() returns an Image type.
var image = texture.get_data()
```

I won't die on this hill though :)